artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/PriorityLinkedListImpl.java

```diff
@@ -54,6 +54,9 @@ public PriorityLinkedListImpl(final int priorities, Comparator<E> comparator) {
    }
 
    private void checkHighest(final int priority) {
+      for (int i = levels.length - 1; i >= 0 && levels[i].size() == 0; i--) {
```

Comment:
Had a look at this and I think this loop is probably not needed if the check in the loop in the iterator remove uses the correct exit terms.  Currently in the remove it scan s for the next highest but it is checking using the wrong variable and always reading from the same collection (index) 

`         for (int i = index; i >= 0 && levels[index].size() == 0; i--) {
            highestPriority = i;
         }
`

When it should be using the loop variable 'i' so that it exists as soon as a non-empty collection is found.

Reply:
Great catch! You're absolutely right. I looked at that code a lot, and my eyes just skipped right over that. Fixed.